### PR TITLE
Add Batch APIs

### PR DESCRIPTION
**Pull request by birdstorm**
Add following batch APIs to the client.

- BatchGet
- BatchScan
- BatchDelete

Signed-off-by: birdstorm <samuelwyf@hotmail.com>

### CODE REVIEW

**Review by marsishandsome — APPROVED** [review]
LGTM

File: src/main/java/org/tikv/common/util/Batch.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 PingCAP, Inc.
```
Comment (marsishandsome) [inline src/main/java/org/tikv/common/util/Batch.java:2]: 2021？

File: src/main/java/org/tikv/common/TiSession.java [hunk 2]
```diff
@@ -59,7 +59,9 @@
   private volatile Catalog catalog;
   private volatile ExecutorService indexScanThreadPool;
   private volatile ExecutorService tableScanThreadPool;
+  private volatile ExecutorService batchGetThreadPool;
```
Comment (marsishandsome) [inline src/main/java/org/tikv/common/TiSession.java:62]: https://github.com/tikv/client-java/blob/master/src/main/java/org/tikv/common/KVClient.java#L137 should use this thread pool

File: src/main/java/org/tikv/raw/RawKVClient.java [hunk 3]
```diff
@@ -280,21 +359,67 @@ private void appendBatches(
...
+          () -> {
+            BackOffer singleBatchBackOffer = ConcreteBackOffer.create(backOffer);
+            try (RegionStoreClient client = clientBuilder.build(batch.region); ) {
+              return client.rawBatchGet(singleBatchBackOffer, batch.keys);
+            } catch (final TiKVException e) {
+              // TODO: any elegant way to re-split the ranges if fails?
+              singleBatchBackOffer.doBackOff(BackOffFunction.BackOffFuncType.BoRegionMiss, e);
+              logger.warn("ReSplitting ranges for BatchGetRequest");
+              // recursive calls
+              batchGet(singleBatchBackOffer, batch.keys);
```
Comment (marsishandsome) [inline src/main/java/org/tikv/raw/RawKVClient.java:385]: big problem here. retry may cause thread pool full.

**Review by marsishandsome — CHANGES_REQUESTED** [review]

File: src/main/java/org/tikv/raw/RawKVClient.java (hunk 3 above)
Comment (birdstorm) [reply src/main/java/org/tikv/raw/RawKVClient.java]: fixed

File: src/main/java/org/tikv/common/util/Batch.java (hunk 1 above)
Comment (birdstorm) [reply src/main/java/org/tikv/common/util/Batch.java]: fixed

File: src/main/java/org/tikv/common/TiSession.java (hunk 2 above)
Comment (birdstorm) [reply src/main/java/org/tikv/common/TiSession.java:62]: fixed

**Review by marsishandsome — APPROVED** [review]
LGTM

**Pull request merged**